docs/README.md

```diff
@@ -75,6 +108,281 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
+    kubectl label namespace bookinfo istio.io/rev=default-v1-21-0
+    ```
+
+7. Deploy bookinfo application
+
+    ```bash
+    kubectl apply -n bookinfo -f https://raw.githubusercontent.com/istio/istio/release-1.22/samples/bookinfo/platform/kube/bookinfo.yaml
+    ```
+
+8. Confirm proxy sidecar injection version match the control plane version
+
+    ```bash
+    kubectl get pod -n bookinfo -l app=details -o yaml | yq '.items[].spec.containers[].image'
+    docker.io/istio/examples-bookinfo-details-v1:1.19.1
+    docker.io/istio/proxyv2:1.21.0
+    ```
+
+9. Update the control plane to a new version
+
+    ```bash
+    cat <<EOF | kubectl apply -f-
+    apiVersion: operator.istio.io/v1alpha1
+    kind: Istio
+    metadata:
+      name: default
+    spec:
+      namespace: istio-system
+      updateStrategy:
+        type: RevisionBased
+        inactiveRevisionDeletionGracePeriodSeconds: 30
+      version: v1.21.2
+    EOF
+    ```
+
+10. Verify the `Istio` and `IstioRevision` resources. There will be a new revision created with the new version.
+
+    ```bash
+    $ kubectl get istio -n istio-system
+      NAME      REVISIONS   READY   IN USE   ACTIVE REVISION   VERSION   AGE
+      default   2           2       1        Healthy           v1.21.2   23m
+    $ kubectl get istiorevision -n istio-system
+      NAME              READY   STATUS    IN USE   VERSION   AGE
+      default-v1-21-0   True    Healthy   True     v1.21.0   27m
+      default-v1-21-2   True    Healthy   False    v1.21.2   4m45s
+    ```
+
+11. Confirm there is two control plane pods running, one for each revision
+
+    ```bash
+    $ kubectl get pods -n istio-system
+      NAME                                      READY   STATUS    RESTARTS   AGE
+      istiod-default-v1-21-0-69d6df7f9c-grm24   1/1     Running   0          28m
+      istiod-default-v1-21-2-7c4f4674c5-4g7n7   1/1     Running   0          6m9s
+    ```
+
+12. Confirm the proxy sidecar version remains the same:
+
+    ```bash
+    $ kubectl get pod -n bookinfo -l app=details -o yaml | yq '.items[].spec.containers[].image'
+      docker.io/istio/examples-bookinfo-details-v1:1.19.1
+      docker.io/istio/proxyv2:1.21.0
+    ```
+
+13. Change the label of the `bookinfo` namespace
+
+    ```bash
+    kubectl label namespace bookinfo istio.io/rev=default-v1-21-2 --overwrite
+    ```
+
+The control plane used by the pods will be updated to the new version only after all workloads are moved to the new revision, and this will happen after the pods are restarted.
+
+14. Delete all the pods in the `bookinfo` namespace
+
+    ```bash
+    kubectl delete pods --all -n bookinfo
+    ```
+
+15. Confirm the new version is used in the sidecars
+
+    ```bash
+    kubectl get pod -n bookinfo -l app=details -o yaml | yq '.items[].spec.containers[].image'
+    docker.io/istio/examples-bookinfo-details-v1:1.19.1
+    docker.io/istio/proxyv2:1.21.2
+    ```
+
+16. Confirm the old control plane deletion
```

Comment:
Add note saying that the old revision (and control plane) are deleted after the grace period specified in `Istio.spec.updateStrategy.inactiveRevisionDeletionGracePeriodSeconds` runs out.

Reply:
@luksa I remember seeing that after we update the label on bookinfo namespace to point to new revision and its restarted, the old Istio revision (and control plane) pods are immediately deleted. i.e., it does not wait for `inactiveRevisionDeletionGracePeriodSeconds` 

Do you think its a bug?